apps/web/components/getting-started/components/ButtonLogic.tsx

```diff
@@ -0,0 +1,14 @@
+import { showToast } from "@calcom/ui";
+
+export const onButtonClick = async (nextStep: () => void, setButtonDisabled: (disabled: boolean) => void) => {
+  setButtonDisabled(true);
+
+  try {
+    await nextStep();
+  } catch (e) {
+    showToast("An error occurred", "error");
+    setTimeout(() => {
+      setButtonDisabled(false);
+    }, 2000);
+  }
+};
```

Comment:
this feels weird

Reply:
To ensure that the button remains disabled after being clicked and is re-enabled only in the event of an error, we will implement a 2-second timeout (timeout can be increased), before allowing the button to be clicked again. This timeout period serves as a preventive measure to mitigate any rapid successive clicks that could potentially exacerbate the error or disrupt the intended functionality. By introducing this brief delay, we aim to maintain a smooth and controlled user experience while handling errors gracefully.